intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,497 @@
+# will be used to specify tensor shapes -- in particular, the value must not be 0 or 1 because
+# many operations have special behavior for tensors with a shape value of 0 or 1.
+
+from torch.export import constrain_as_size, constrain_as_value
+
+def constraints_example4(x, y):
+    b = y.item()
+    constrain_as_value(b, 3, 5)
+    if b >= 3:
+       return x.cos()
+    return x.sin()
+
+exported_constraints_example4 = export(constraints_example4, (torch.randn(3, 3), torch.tensor([4])))
+print(exported_constraints_example4(torch.randn(3, 3), torch.tensor([5])))
+try:
+    exported_constraints_example4(torch.randn(3, 3), torch.randn([2]))
+except Exception:
+    tb.print_exc()
+
+def constraints_example5(x, y):
+    b = y.item()
+    constrain_as_size(b)
+    z = torch.ones(b, 4)
+    return x.sum() + z.sum()
+
+exported_constraints_example5 = export(constraints_example5, (torch.randn(2, 2), torch.tensor([4])))
+print(exported_constraints_example5(torch.randn(2, 2), torch.tensor([5])))
+try:
+    exported_constraints_example5(torch.randn(2, 2), torch.randn([1]))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# Custom Ops
+# ----------
+# ``torch.export`` can export PyTorch programs with custom operators.
+#
+# NOTE: the API for registering custom ops is still under active development
+# and may change without notice.
+#
+# Currently, the steps to register a custom op for use by ``torch.export`` are:
+#
+# - Define the custom op using ``torch.library`` (`reference <https://pytorch.org/docs/main/library.html>`__)
+# as with any other custom op
+
+from torch.library import Library, impl
+
+m = Library("my_custom_library", "DEF")
+
+m.define("custom_op(Tensor input) -> Tensor")
+
+@impl(m, "custom_op", "CompositeExplicitAutograd")
+def custom_op(x):
+    print("custom_op called!")
+    return torch.relu(x)
+
+# - Define a ``"Meta"`` implementation of the custom op that returns an empty
+# tensor with the same shape as the expected output
+
+@impl(m, "custom_op", "Meta")
+def custom_op_meta(x):
+    return torch.empty_like(x)
+
+# - Call the custom op from the code you want to export using ``torch.ops``
+
+def custom_op_example(x):
+    x = torch.sin(x)
+    x = torch.ops.my_custom_library.custom_op(x)
+    x = torch.cos(x)
+    return x
+
+# - Export the code as before
+
+exported_custom_op_example = export(custom_op_example, (torch.randn(3, 3),))
+exported_custom_op_example.graph_module.print_readable()
+print(exported_custom_op_example(torch.randn(3, 3)))
+
+# Note in the above outputs that the custom op is included in the exported graph.
+# And when we call the exported graph as a function, the original custom op is called,
+# as evidenced by the ``print`` call.
+
+######################################################################
+# ExportDB
+# --------
+# ``torch.export`` will only ever export a single computation graph from a PyTorch program. Because of this requirement,
```

Comment:
as noted above, I think this is a really important point and should be earlier in the tutorial.